docs/docs/developers/contracts/syntax/functions/public_private_unconstrained.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Public, Private, and Unconstrained Functions
+---
+
+This page explains the three types of functions that exist on Aztec - public, private, and unconstrained. For a deeper dive into how these functions work under the hood, check out the [Inner Workings](./inner_workings.md) page.
+
+## `Public` Functions
+
+A public function is executed by the sequencer and has access to a state model that is very similar to that of the EVM and Ethereum. Even though they work in an EVM-like model for public transactions, they are able to write data into private storage that can be consumed later by a private function.
+
+To create a public function you can annotate it with the `#[aztec(public)]` attribute. This will make the [public context](../context.mdx) available within your current function's execution scope.
+
+#include_code set_minter /yarn-project/noir-contracts/contracts/token_contract/src/main.nr rust
+
+## `Private` Functions
+
+As alluded to earlier, a private function operates on private information, and is executed by the user. To tell the compiler that this is the kind of function we are creating annotate it with the `#[aztec(private)]` attribute. This will make the [private context](../context.mdx#private-context-broken-down) available within your current function's execution scope.
```

Comment:
```suggestion
A private function operates on private information, and is executed by the user. Annotate the function with the `#[aztec(private)]` attribute to tell the compiler it's a private function. This will make the [private context](../context.mdx#private-context-broken-down) available within the function's execution scope.
```